src/components/icon.scss

```diff
@@ -0,0 +1,25 @@
+// * Icon component
+// *
+// * 1. Fine tune the positioning of the icon
+// * 2. Fill with currentColor by default
+// *
+
+.Icon {
+  position: relative; // 1
+  margin: 0;
+  display: inline-block;
+  width: 1em;
+  height: 1em;
+  font-size: 1em;
+  vertical-align: middle;
+  top: -1px; // 1
+  fill: currentColor; // 2
+}
+
+// * Modifier -- large icon relative to parent text
+
+$Icon--large-size: 1.25em;
+
+.Icon--large {
```

Comment:
I keep going back and forth on this, but from a design standpoint have a hard time seeing modifiers being used with icons.  Typically that icon size is going to be set by the component using it (mostly at 90%), but in the cases where it needs a specific size I have a hard time seeing that being standardized in any way... at least right now, the more we move to design scales that could change.